lib/gc/Dialect/CPURuntime/Transforms/CPURuntimeToLLVM.cpp

```diff
@@ -109,6 +109,78 @@ class PrintfRewriter : public ConvertOpToLLVMPattern<PrintfOp> {
   }
 };
 
+template <typename OpType, typename OpAdaptor, const char *allocFuncName>
+class AlignedAllocRewriterBase : public ConvertOpToLLVMPattern<OpType> {
+public:
+  using ConvertOpToLLVMPattern<OpType>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(OpType runtimeOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    Operation *op = runtimeOp;
+    MLIRContext *context = op->getContext();
+    auto moduleOp = op->getParentOfType<ModuleOp>();
+    auto loc = op->getLoc();
+    MemRefType memRefType =
+        dyn_cast<MemRefType>(op->getResults().front().getType());
```

Comment:
Shall we validate the `memRefType` from `dyn_cast` here?

Reply:
Thanks for the suggestion, validated.